src/hooks/use-update-demo-site.tsx

```diff
@@ -35,10 +36,37 @@ export const DemoSiteUpdateProvider: React.FC< DemoSiteUpdateProviderProps > = (
 			setUpdatingSites( ( prev ) => new Set( prev ).add( localSite.id ) );
 
 			try {
-				const { archiveContent } = await getIpcApi().archiveSite( localSite.id, 'zip' );
-				const file = new File( [ archiveContent ], 'loca-env-site-1.zip', {
-					type: 'application/zip',
-				} );
+				const { archivePath, archiveSizeInBytes } = await getIpcApi().archiveSite(
+					localSite.id,
+					'zip'
+				);
+
+				if ( archiveSizeInBytes > DEMO_SITE_SIZE_LIMIT_BYTES ) {
+					alert(
+						sprintf(
+							__(
+								'The site exceeds the maximum size of %dGB. Please remove some files and try again.'
+							),
+							DEMO_SITE_SIZE_LIMIT_GB
+						)
+					);
```

Comment:
We could consider updating the code to use `showErrorMessageBox` or `showMessageBox` instead of `alert`.

Something like:

```suggestion
					getIpcApi().showErrorMessageBox( {
						title: __( 'Updating demo site failed' ),
						message: sprintf(
							__(
								'The site exceeds the maximum size of %dGB. Please remove some files and try again.'
							),
							DEMO_SITE_SIZE_LIMIT_GB
						),
					} );
```

<img width="1012" alt="Screenshot 2025-01-17 at 14 37 51" src="https://github.com/user-attachments/assets/4f9aae53-18ea-4b4a-9c0b-bbd284adb89d" />

Same comment for the message displayed in use-archive-site.

Reply:
Yep! I agree. I did wonder about it whether we choose what to use on the basis of context.